src/OpenStudy/relations.jl

```diff
@@ -119,6 +120,74 @@ function _get_relation(source::String, target::String, relation_type::RelationTy
     return _RELATIONS[source][(target, relation_type)]
 end
 
+function _get_target_index_from_relation(data::Data, source::String, target::String, source_index::Integer, relation_attribute::String)
+    source_element = data.raw[source][source_index]
+    target_index = _get_index(data.data_index,source_element[relation_attribute])
+    return target_index[2]
+end
+
+function _get_sources_indices_from_relations(data::Data, source::String, target::String, target_id::Integer, relation_attribute::String,)
+    possible_elements = data.raw[source]
+
+    indices = Vector{Int32}()
+    for (index,element) in enumerate(possible_elements)
+        if haskey(element, relation_attribute)
+            if element[relation_attribute] == target_id
+                append!(indices, index)
```

Comment:
Use `push!` instead of `append!`. In Julia, `push!` works like Python's `list.append` and `append!` works like `list.extend`!

Reply:
Oh, thanks for the tip!